### ISSUE TITLE
Named routes?

### ISSUE THREAD

**Issue by johnpaulkiser** [yesmeck/remix-routes#92]
I would love to be able to name a route (similar to django & laravel). Route names are used in these other mature frameworks to make url refactors easier.

I'm not sure exactly what the API would look like, but do you have any thoughts about this idea?

**Comment by airtonix** [comment]
I can chime in here having had a decent amount of experience with Django.

When I moved from Django to Nodejs development, this was the one thing that exasperated me about the nodejs community; they did not know this would be a scaling problem.

However; Counterpoint: Typescript makes this irrelevant.

Django needs named routes mainly because the type safety is so poor in Python (yes even with the recent edition)

RemixRoutes will generate type for you so that refactors will throw type errors if you miss refactoring some of your routes.

tldr: no needed because typescript typechecking will catch mismatches.

**Comment by johnpaulkiser** [comment]
I also imagine any sort of development with this package is probably on hold while we wait for react-router <-> remix merge to see their approach to.
